### ISSUE TITLE
[App Support] beardedspice no longer detects applicable tabs

### ISSUE THREAD

**Issue by peterschussheim** [beardedspice/beardedspice#622]
Hi!

First, thank you for this app, it has been very useful (when it works).  Starting 2-3 days ago, I lost all ability to control tabs playing music.  Bearded Spice just says 'No Applicable tabs open'. I have tried completely uninstalling the app with no success.

My system is  MacBookPro12,1 running latest macOS build.

Thank you,
Peter

**Comment by 1ps0** [comment]
[See if this helps](https://github.com/beardedspice/beardedspice/wiki/Wont-Work-issue-after-Safari-Update-(9.1.1-and-later)).

**Comment by peterschussheim** [comment]
Hi, thanks for the link, unfortunately it did not help.

**Comment by Mauin** [comment]
Seeing the same with Google Chrome `Version 58.0.3029.41 (Official Build) beta (64-bit)`. Let me know if there are some logs I can attach that would help debug this.

**Comment by 1ps0** [comment]
@Mauin can you do me a favor and check if that build has applescript enabled?

In Terminal, `sdef /Applications/Chrome-beta-or-whatever.app`. It should say right away if its even available.

**Comment by lirao** [comment]
I have this:
```
→ sdef /Applications/Google\ Chrome.app | grep -i applescript
				<cocoa key="appleScriptWindows"/>
			<cocoa class="WindowAppleScript"/>
			<cocoa class="TabAppleScript"/>
			<cocoa class="BookmarkFolderAppleScript"/>
			<cocoa class="BookmarkItemAppleScript"/>
```

**Comment by Mauin** [comment]
@1ps0 As @lirao also mentioned in his comment I can see the same mentions to applescript in the output. Not sure if that is enough or if I should look for more mentions.

**Comment by 1ps0** [comment]
@lirao @Mauin Thanks for the verification.

What it means is that Google didn't lock down Chrome (beta or otherwise) and there's probably a straightforward fix. I'll be addressing a number of things this weekend, I'll make this top priority.

**Comment by 1ps0** [comment]
@Mauin While I'm at it, can you verify two things for me?

1. That `Accessibility` is enabled for BeardedSpice. Under `System Preferences > Security & Privacy > Privacy > Accessibility`

2. That the BeardedSpice app is being run from the `/Applications` directory. Since Sierra has issues with apps not launched from there.

**Comment by Mauin** [comment]
@1ps0 Yep, in my case `Accessibility` is enabled for BeardedSpice and it's being run from the `/Applications` directory.

**Comment by lirao** [comment]
I'm still on El Capitan!
And yep, Accessibility is enabled.

**Comment by 1ps0** [comment]
@Mauin for you specifically it is actually the Chrome Beta breaking. I just had a chance to properly install/test/dump the headers and compare them. Absolutely doesn't work at all for ChromeBeta (58 doesnt matter I think).

Proof is in the pudding. Right from the top, instead of `ChromeApplication` defined as our core interface object, we have `ChromeBetaApplication`. Just to change enough to accept this would require a recompile and redeploy of the app.

You can dump the header fully (not the binary test I had you do earlier) with `sdef /Applications/Google Chrome Betawhatever.app | sdp -fh --basename ChromeBeta.h` and check it against the [normal chrome headers](https://github.com/beardedspice/beardedspice/blob/master/BeardedSpice/Tabs/Chrome.h).

Currently on vacation and dont have access to a variety of machines. My dev laptop is El Capitan, and I cannot repro the other issue yet.

**Comment by mitchell486** [comment]
@lirao @Mauin, @1ps0  Is correct. I had Chrome Beta 58, and I just downgraded to 57 Stable. It is now working on macOS 10.12.4 (16E195). Thank you!

**Comment by Mauin** [comment]
That's good to know. But are these different headers something that have recently been introduced in the Chrome Beta? I have been using the Beta channel for a long time now and Beardedspice just recently began having issues with it. Or is this change in Beta something that will come to the stable channel of Chrome as well at some point?

**Comment by Mauin** [comment]
Funnily enough, I just tried this with the Google Chrome Canary Channel (currently: 59.0.3071.0) and Beardedspice works as a charm again.

**Comment by peterschussheim** [comment]
Still not working for me on canary :(

**Comment by 1ps0** [comment]
@Mauin this is basically the argument for not supporting betas or nightlies. They change so often and so fast, almost on the nightly whims of the dev teams.

**Comment by Mauin** [comment]
Yep, totally agree @1ps0. Especially the Canary/Nightly channel should not be a focus of support. With Beta one could argue that it will potentially forthshadow any issues that might be promoted to hit the stable channel.

**Comment by lirao** [comment]
Thanks! Switched back to stable channel and it worked right away!

**Comment by peterschussheim** [comment]
@1ps0 I have very little experience with macOS development but would love to help build a solution for these nightly builds of Chrome.  If you have a moment, can you share more specifically why this is not a trivial fix?  I do not understand your comment above about normal chrome headers versus canary headers.
